### ISSUE TITLE
torch.nn.functional.conv1 return different values when the batch size of the input tensor is different

### ISSUE THREAD

**Issue by SoonSYJ** [pytorch/pytorch#91612]
### 🐛 Describe the bug

🐛 Describe the bug
You can reproduce the bug using the follow scripts.
`
import torch
import matplotlib.pyplot as plt
def imshow(x, **kwargs):
    if isinstance(x, torch.Tensor):
        x = x.cpu().detach()
    plt.imshow(x, aspect='auto', origin='lower', **kwargs)

w_1 = torch.nn.Conv1d(
    256,
    1024,
    3,
    stride=1,
    padding="valid",
    padding_mode="reflect",
    bias=False,
)
w_1.eval()

xs = torch.rand(32, 25, 256)
xss = xs.transpose(-1,-2)

x1 = xss[:1].clone() 
x2 = xss[:20].clone()
_x1 = w_1(x1)
_x2 = w_1(x2)

print(_x1.shape, _x2.shape)
imshow(_x1[0] - _x2[0])
print(abs(_x1[0] - _x2[0]).max())
plt.colorbar()
`
Result:
torch.Size([1, 1024, 23]) torch.Size([20, 1024, 23])
tensor(8.9407e-07)   # this error should be zero
![image](https://user-images.githubusercontent.com/11547738/210318176-6b9b3f5d-8acf-46e6-8583-e5d2d4d5e191.png)

### Versions

PyTorch version: 1.13.0
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
Clang version: Could not collect
CMake version: version 3.14.7
Libc version: glibc-2.27

Python version: 3.8.15 | packaged by conda-forge | (default, Nov 22 2022, 08:49:35)  [GCC 10.4.0] (64-bit runtime)
Python platform: Linux-5.4.0-135-generic-x86_64-with-glibc2.10
Is CUDA available: True
CUDA runtime version: 11.6.55
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: 
GPU 0: NVIDIA GeForce RTX 3090
GPU 1: NVIDIA GeForce RTX 3090
GPU 2: NVIDIA GeForce RTX 3090
GPU 3: NVIDIA GeForce RTX 3090

Nvidia driver version: 515.65.01
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.23.4
[pip3] torch==1.13.0
[pip3] torchaudio==0.13.0
[pip3] torchvision==0.14.0
[conda] blas                      1.0                         mkl  
[conda] ffmpeg                    4.3                  hf484d3e_0    pytorch
[conda] mkl                       2021.4.0           h06a4308_640  
[conda] mkl-service               2.4.0            py38h95df7f1_0    conda-forge
[conda] mkl_fft                   1.3.1            py38h8666266_1    conda-forge
[conda] mkl_random                1.2.2            py38h1abd341_0    conda-forge
[conda] numpy                     1.23.4           py38h14f4228_0  
[conda] numpy-base                1.23.4           py38h31eccc5_0  
[conda] pytorch                   1.13.0          py3.8_cuda11.7_cudnn8.5.0_0    pytorch
[conda] pytorch-cuda              11.7                 h67b0de4_0    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchaudio                0.13.0               py38_cu117    pytorch
[conda] torchvision               0.14.0               py38_cu117    pytorch

cc @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

**Comment by helson73** [comment]
Pytorch choose cudnn convolution algorithm using benchmark and heuristics. Mostly that depend on input tensors' shape and different algorithms are expected to yield slightly different numerical results.

**Comment by albanD** [comment]
Hi,

Any error below 1-e6 is basically zero for fp32 precision. You can see more details at https://pytorch.org/docs/stable/notes/numerical_accuracy.html

Closing as expected behavior.
